src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    /**
+     * The method for addEvents
+     * @param description of events task
+     * @param list Tasklist where task is going to be stored in
+     * @param start the starting time of the event
+     * @param end the ending time of the event
+     */
+    private static void addEvents(String description, TaskList list, LocalDateTime start, LocalTime end) {
+        Events evItem = new Events(description, start, end);
+        list.addTask(evItem);
+        System.out.println("\tGot it. I have added this task:\n " + "\t" + evItem.toString());
+        System.out.printf("\tNow you have %d tasks in the list.\n", list.size());
+    }
+
+    /**
+     * The method for addDeadline
+     * @param description of deadline task
+     * @param list tasklist where task going to be stored in
+     * @param doneBy the time of deadline task
+     */
+
+    private static void addDeadline(String description, TaskList list, LocalDateTime doneBy) {
+        Deadline dlItem = new Deadline(description, doneBy);
+        list.addTask(dlItem);
+        System.out.println("\tGot it. I have added this task:\n " + "\t" + dlItem.toString());
+        System.out.printf("\tYou have %d tasks in the list.\n\n", list.size());
+    }
+
+    /**
+     * The method for delete
+     * @param number the
+     * @param list
+     */
+    public static void delete(String number, TaskList list) {
+        Task removed = list.get(Integer.parseInt(number.split(" ")[1]) );
+        int nummbering = Integer.parseInt(number.split(" ")[1]);
+        list.removeTask(nummbering);
+        Ui.Underline();
+        Ui.showDeleteMessage();
+        System.out.println(String.format("\t%s removed", removed.toString()));
+        System.out.println("\tNow you have " + list.size() + " tasks in the list.");
+    }
+    public static void findTask(TaskList list, String wantedTask) {
+        ArrayList<Task> containingList = new ArrayList<>();
+        for (int i = 1; i <= list.size(); i++) {
+            Task task = list.get(i);
+            if (task.contains(wantedTask)) {
+                containingList.add(task);
+            }
+        }
+        if (containingList.size() == 0) {
+            System.out.println("\tSorry you have no matching tasks.\n");
+            return;
+        }
+        System.out.println("\tHere are the matching tasks in your list:");
+        for (int i = 1; i <= containingList.size(); i++) {
+            System.out.println("\t" + i + "." + containingList.get(i - 1));
+
+
+        }
+        System.out.println();
+
+    }
+
+
+    /**
+     * The method for parse which makes sense of the user input and execute the correct actions
+     * @param instct the reader input to parse
+     * @param list Tasklist  which actions are to be performed on
+     * @return int of -1 if bye instruct is parsed else 1 is return to signal program to continue running
+     */
+    public static int parse(String instct, TaskList list) throws  IOException  {
```

Comment:
This method seems to be quite long. Perhaps break it up into smaller methods (such as a method for each command, or a new class for each command) would allow for easier code maintenance in the future?